pkg/reconciler/bitbucketsource/resources/bitbucket_client.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/knative/pkg/logging"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const (
+	defaultBaseUrl   = "https://api.bitbucket.org/2.0/"
+	defaultUserAgent = "go-bitbucket"
+	mediaTypeJson    = "application/json"
+)
+
+// Hook struct to marshal/unmarshal BitBucket requests/responses.
+type Hook struct {
+	URL         string   `json:"url,omitempty"`
+	Description string   `json:"description,omitempty"`
+	Events      []string `json:"events,omitempty"`
+	Active      bool     `json:"active,omitempty"`
+	UUID        string   `json:"uuid,omitempty"`
+}
+
+// Client struct used to send http.Requests to BitBucket.
+type Client struct {
```

Comment:
Please move all of this source to something like `github.com/knative/eventing-sources/pkg/bitbucket/client` or something else that makes sense. It is not part of the reconciler code and might one day be vendor'ed.

Reply:
Done. I'm not very convinced of the folder, but I haven't come up with a better location, so I just put it in `github.com/knative/eventing-sources/pkg/bitbucket/client`